allel/util.py

```diff
@@ -145,6 +146,36 @@ def ensure_square(dist):
     return dist
 
 
+def mask_inaccessible(is_accessible, ac, pos):
+    """ This function removes any positions that are not accessible from the
+    allele counts Array and the vector of variant positions.
+
+    Parameters
+    ----------
+    is_accessible : array_like, bool, shape (len(contig),), optional
+        Boolean array indicating accessibility status for all positions in the
+        chromosome/contig.
+    ac : array_like, int, shape (n_variants, n_alleles)
+        Allele counts array.
+    pos : array_like, int, shape (n_items,)
+        Variant positions, using 1-based coordinates, in ascending order.
+
+    Returns
+    -------
+    ac : array_like, int, shape (n_variants, n_alleles)
+        Masked allele counts array.
+    pos : array_like, int, shape (n_items,)
+        Masked variant positions, using 1-based coordinates, in ascending order.
+
+    """
```

Comment:
Should check if `is_accessible` is the same length as `pos` and `ac`, raising a ValueError if not?

Reply:
I mean that `is_accessible` should be the length of the *genome* (ie. the length of `stop` minus `start` or something).

Reply:
and there are no `pos` that are greater than the length of `is_accessible` (plus one?)

Reply:
I don't think we should check if `len(is_accessible) == stop-start+1` because you might want to delimit your diversity calculation to a subregion of the genome. But agree checking if any pos is greater than `len(is_accessible)` is worthwhile.